src/main/java/org/fundacionjala/core/selenium/interaction/WebDriverManager.java

```diff
@@ -0,0 +1,74 @@
+package org.fundacionjala.core.selenium.interaction;
+
+import org.fundacionjala.core.selenium.browsers.Browser;
+import org.fundacionjala.core.selenium.browsers.BrowserFactory;
+import org.fundacionjala.core.selenium.browsers.BrowserParser;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.ui.WebDriverWait;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * [MR] Class in charge to mage the WebDriver provided by the BrowserFactory.
+ */
+public final class WebDriverManager {
+
+    private static WebDriverManager webDriverManager;
+    private static String browserName;
+    private final WebDriver webDriver;
+    private final WebDriverWait webDriverWait;
+
+
+    /**
+     * If webDriverManager object was not created before it create a new one,
+     * otherwise return the created.
+     * @return webDriverManager
+     */
+    public static WebDriverManager getInstance() {
+        if (webDriverManager == null) {
+            webDriverManager = new WebDriverManager();
+        }
+        return webDriverManager;
+    }
+
+    private WebDriverManager() {
+        webDriver = BrowserFactory.getWebDriver(browserName);
+        Browser browser = BrowserParser.getBrowsersMap().get(browserName);
+        webDriver.manage().window().maximize();
+        webDriver.manage().timeouts().
+                implicitlyWait((long) browser.getImplicitWaitingSeconds(), TimeUnit.SECONDS);
+        webDriverWait = new WebDriverWait(webDriver, (long) browser.getExplicitWaitingSeconds(),
+                (long) browser.getSleepingTimeMills());
```

Comment:
I think the method " browser.getImplicitWaitingSeconds()" should be return a long, so we avoid the casting (long)

Reply:
the implicitWaitingSeconds is defined as a double value in Browser this is the reason of the cast to long; do you want that I change all attributes to long?

Reply:
Yep, please change to long.